sheepdog/utils/__init__.py

```diff
@@ -439,8 +439,19 @@ def generate_s3_url(host, bucket, program, project, uuid, file_name):
     if not bucket.endswith('/'):
         bucket += '/'
 
-    key = '{}/{}/{}/{}'.format(program, project, uuid, file_name)
-    return host + bucket + key
+    return host + bucket + generate_s3_key(program, project, uuid, file_name)
```

Comment:
plz do `'/'.join(host, bucket, key)` and remove lines appending the `/` above

Reply:
sure